packages/console/src/components/Navigation/DefaultAppBarContent.tsx

```diff
@@ -33,13 +28,53 @@ interface DefaultAppBarProps {
 export const DefaultAppBarContent = (props: DefaultAppBarProps) => {
   const [platformVersion, setPlatformVersion] = React.useState('');
   const [consoleVersion, setConsoleVersion] = React.useState('');
+  const {
+    isMobileNav,
+    openSideNav,
+    closeSideNav,
+    isSideNavOpen,
+    isLayoutHorizontal,
+    showMobileNav,
+    hideMobileNav,
+  } = React.useContext(TopLevelLayoutContext);
+
   const commonStyles = useCommonStyles();
-  const styles = useStyles();
 
   const isFlagEnabled = useFeatureFlag(FeatureFlag.OnlyMine);
   const { adminVersion } = useAdminVersion();
   const isGAEnabled = env.ENABLE_GA === 'true' && env.GA_TRACKING_ID !== '';
 
+  const handleSideNavToggle = React.useCallback(() => {
+    if (isSideNavOpen) {
+      closeSideNav();
+    } else {
+      openSideNav();
+    }
+  }, [isSideNavOpen, openSideNav, closeSideNav]);
+
+  const theme = useTheme();
+
+  // Enable / Disable mobile nav behaviour based on screen size
+  React.useLayoutEffect(() => {
+    const handleResize = () => {
+      if (window.innerWidth < theme.breakpoints.values.md) {
+        if (!isMobileNav) {
```

Comment:
There is an unnecessary if statement inside another if. Please group the condition using &&.

You should also return early if an if statement is not satisfied since it prevents code nesting and easy to read